daml-lf/scenario-interpreter/src/main/scala/com/digitalasset/daml/lf/ScenarioRunner.scala

```diff
@@ -399,6 +401,9 @@ object ScenarioRunner {
       commitLocation = location,
       limits = interpretation.Limits.Lenient,
     )
+
+    // TODO (drsk) propagate errors back to submitter
+    val discTable = Engine.buildDiscTable(ledgerMachine, disclosures)
```

Comment:
You are missing some validation somewhere. Probably when committing the transaction to the scenario ledger. If I pass garbage disclosed contracts they need to get rejected. I’m fine with doing that in a separate PR if you want but let’s track it somewhere.

Reply:
I added a ticket to track this for a next PR.